### ISSUE TITLE
python3.6: Multiprocessing.pool and negative return code

### ISSUE THREAD

**Issue by viniul** [amoffat/sh#458]
If an `sh.Command` call returns a non-zero exit code it seems the multiprocessing module can't pickle the exception. Consider the following code:
```

#!/usr/bin/env python3
from multiprocessing import Pool
import sh


def f( x): 
    sh.Command("./exit.sh")()

p = Pool(5)
p.map(f, [(x, ) for x in range(1000)])
p.close()
p.join()
```
With `exit.sh` consisting of the line `exit 1`.

This script throws the following exception and apparently freezes afterwards:
```


$ ./test.py 
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib64/python3.6/multiprocessing/pool.py", line 463, in _handle_results
    task = get()
  File "/usr/lib64/python3.6/multiprocessing/connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
TypeError: __init__() missing 2 required positional arguments: 'stdout' and 'stderr'
```

**Comment by Avery-Whitaker** [comment]
Currently when this happens the program hangs. If solving this issue is impossible, can it at least be made to terminate?

**Comment by amoffat** [comment]
This is fixed by e3c49895cca67435e7d5370f36d5e7fd16f9313d and will go out with the next release.
